elfloader-tool/src/arch-riscv/crt0.S

```diff
@@ -41,6 +48,21 @@ bootstack_secondary_cores:
 
 .text
 
+switch_hart:
+/* Check HSM extension exists so we can use that to switch
+   otherwise rely on the legacy extension */
+  jal spi_probe_extension
+  li s0, 1
+  blt x1, s0, secondary_harts
```

Comment:
please add a bit info about what value  is returned by spi_probe_extension and why the value is compared with 1.

Reply:
so if the extension is not supported,  the current hart is not hart1, and SMP is not enabled, the hart just spins. Is that the expected results?

Reply:
Yes that's right. BBL worked like this. If using OPENSBI this code path shouldn't be used.